src/Aeson.purs

```diff
@@ -23,74 +23,125 @@
+import Foreign.Object (Object)
 import Foreign.Object as FO
+import Partial.Unsafe (unsafePartial)
+import Prim.Row as Row
+import Prim.RowList as RL
+import Record as Record
+import Type.Prelude (Proxy(Proxy))
 
--- | A piece of JSON where all numbers are extracted away into `NumberIndex`.
-newtype Aeson = Aeson { json :: Json, index :: NumberIndex }
+-- | A piece of JSON where all numbers are replaced with their indexes
+newtype AesonPatchedJson = AesonPatchedJson Json
 
-getJson :: Aeson -> Json
-getJson (Aeson { json }) = json
+-- | A piece of JSON where all numbers are extracted into `NumberIndex`.
+newtype Aeson = Aeson { patchedJson :: AesonPatchedJson, numberIndex :: NumberIndex }
 
 -- | A list of numbers extracted from Json, as they appear in the payload.
 type NumberIndex = Array String
 
 class DecodeAeson (a :: Type) where
   decodeAeson :: Aeson -> Either JsonDecodeError a
 
-instance DecodeAeson Int where
-  decodeAeson aeson@(Aeson { index }) = do
-    -- Numbers are replaced by their index in the array.
-    ix <- decodeAesonViaJson aeson
-    numberStr <- note MissingValue (index Array.!! ix)
-    note MissingValue $ Int.fromString numberStr
+-------- Parsing: String -> Aeson --------
 
-instance DecodeAeson BigInt where
-  decodeAeson aeson@(Aeson { index }) = do
-    -- Numbers are replaced by their index in the array.
-    ix <- decodeAesonViaJson aeson
-    numberStr <- note MissingValue (index Array.!! ix)
-    note MissingValue $ BigInt.fromString numberStr
+foreign import parseJsonExtractingIntegers
+  :: String
+  -> { patchedPayload :: String, numberIndex :: NumberIndex }
 
-instance DecodeAeson String where
-  decodeAeson = decodeAesonViaJson
+parseJsonStringToAeson :: String -> Either JsonDecodeError Aeson
+parseJsonStringToAeson payload = do
+  let { patchedPayload, numberIndex } = parseJsonExtractingIntegers payload
+  patchedJson <- note MissingValue $ hush $ AesonPatchedJson <$> jsonParser patchedPayload
+  pure $ Aeson { numberIndex, patchedJson }
 
-instance DecodeAeson Number where
-  decodeAeson = decodeAesonViaJson
+-------- Json <-> Aeson --------
 
-instance DecodeAeson Json where
-  decodeAeson = decodeAesonViaJson
+-- | Replaces indexes in the Aeson's payload with stringified
+--   numbers from numberIndex.
+--   Given original payload of: `{"a": 10}`
+--   The result will be an Json object representing: `{"a": "10"}`
+toStringifiedNumbersJson :: Aeson -> Json
+toStringifiedNumbersJson = fix \_ ->
+  caseAeson
+    { caseNull: const jsonNull
+    , caseBoolean: encodeBoolean
+    , caseNumber: encodeString
+    , caseString: encodeString
+    , caseArray: map toStringifiedNumbersJson >>> fromArray
+    , caseObject: map toStringifiedNumbersJson >>> fromObject
+    }
 
-instance DecodeAeson Aeson where
-  decodeAeson = pure
+-- | Recodes Json to Aeson.
+--   NOTE. The operation is costly as its stringifies given Json
+--         and reparses resulting string as Aeson.
+jsonToAeson :: Json -> Aeson
```

Comment:
Why do we need this? Just for testing?

Reply:
I think I use it in the other PR for errors reports